### DISCUSSION TITLE
How can I create a smarter edge path?

### DISCUSSION THREAD

**Discussion by binhtran04** [xyflow/xyflow#3195] [Q&A]
Hi,
How can I achieve something like the image below, where an edge connect from the right handle of a node (the **blue one**) goes back to the left handle of a previous node?

![image](https://github.com/wbkd/react-flow/assets/15939436/acb0a63b-7c08-413f-9f6e-0427ab8b88ac)

I tried all the edge types but could not find a suitable one. Below is an example using `smoothstep`. The edge goes under other nodes, which is very difficult to see

![image](https://github.com/wbkd/react-flow/assets/15939436/8d1d9033-ebdb-4bcc-8c47-15b28e9cacd0)

**Comment by binhtran04** [comment]
It seems that react-flow does not have yet a smart routing strategy. Closed the discussion

**Comment by tisoap** [comment]
I'm the creator of [react-flow-smart-edge](https://github.com/tisoap/react-flow-smart-edge) , a library that uses using A* pathfinding to route edges around other nodes, avoiding intersections. It has smart equivalents for every type of edge React Flow has. It's been updated to work with React Flow 12.
